msticpy/data/uploaders/splunk_uploader.py

```diff
@@ -0,0 +1,252 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""Splunk Uploader class."""
+from pathlib import Path
+from typing import Any
+from tqdm.notebook import tqdm
+import pandas as pd
+from pandas.io.parsers import ParserError
+
+from .uploader_base import UploaderBase
+from ..._version import VERSION
+from ..drivers.splunk_driver import SplunkDriver
+from ...common.exceptions import MsticpyConnectionError, MsticpyUserError
+
+__version__ = VERSION
+__author__ = "Pete Bryan"
+
+
+class SplunkUploader(UploaderBase):
+    """Uploader class for Splunk."""
+
+    def __init__(self, username: str, host: str, password: str, **kwargs):
```

Comment:
You might want to re-use the code (maybe we factor it out into some common splunk module) from the Splunk driver. This uses either params, connection string or config values. Again, can be done in future.